src/actions/AddTokenActions.js

```diff
@@ -48,6 +49,12 @@ export const addTokenAsync = async (
 ) => {
   // create modified object structure to match metaTokens
   const newTokenObj: CustomTokenInfo = WALLET_ACTIONS.assembleCustomToken(currencyName, currencyCode, contractAddress, denomination)
+  const plugins: Object = state.ui.settings.plugins
+  const allCurrencyInfos = plugins.allCurrencyInfos
+  const allSupportedParentCurrencies = allCurrencyInfos.map(info => info.currencyCode)
+  if (allSupportedParentCurrencies.find(item => item === currencyCode)) {
+    throw new Error(s.strings.error_token_exists)
+  }
```

Comment:
Right, this is one place custom tokens can get into the system - from the user at runtime.